sql/step_a111_settings.sql

```diff
@@ -0,0 +1,95 @@
+-- Drop the existing table
+DROP TABLE IF EXISTS OH_SETTINGS;
+
+-- Create settings table structure
+CREATE TABLE OH_SETTINGS (
+  SETT_ID INT(11) NOT NULL AUTO_INCREMENT COMMENT 'Default generated ID',
+  SETT_CODE VARCHAR(50) NOT NULL,
+  SETT_VALUE_TYPE ENUM('bool', 'number', 'text', 'select') NOT NULL DEFAULT 'text',
+  SETT_VALUE_OPTIONS VARCHAR(500) NULL DEFAULT NULL COMMENT "Comma-separated list of possible values, in case of type 'select'",
+  SETT_DEFAULT_VALUE VARCHAR(255) NOT NULL,
+  SETT_VALUE VARCHAR(255) NOT NULL,
+  SETT_DESCRIPTION VARCHAR(500) NULL,
+  SETT_CREATED_BY VARCHAR(50) NULL DEFAULT NULL,
+  SETT_MODIFIED_BY VARCHAR(50) NULL DEFAULT NULL,
+  SETT_CREATED_DATE DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP(),
+  SETT_MODIFIED_DATE DATETIME NULL DEFAULT NULL,
+  SETT_DELETED TINYINT(1) NOT NULL DEFAULT 0,
+  SETT_EDITABLE TINYINT(1) NOT NULL DEFAULT 1,
```

Comment:
what this two are meant for?

Reply:
For `SETT_DELETED` I was just following convention, but I think we don't really need soft deletion as `settings` can't be deleted, at least, by the user.

For `SETT_EDITABLE`, I think we should not allow users to change some settings value like `PARAMSURL` and "report file names" (like `VISITSHEET`)

Reply:
all fields are there to be modifiable, otherwise they would be inside the code and not outside. The only setting that might benefit of this is SINGLEUSER by the moment is very sensitive, although I don't think one should write directly in the DB in order to change it.

Also, about the DEFAULT_VALUE: that is meant to take place when the parameter is not specified at all, but this means that the record is not in the DB, so also its default value... the default value will be still hardcoded in the code and this field will be there just as reminder to reset the field, right?

Reply:
> Should I then remove or keep it?

Let's remove it for now as it sound useless

> If you see things differently, explain to me so that I can put it into practice.

We are just brainstorming here, by the moment the issue is not describing every single case.
What do you mean with "just raise an exception and cancel the current operation"? I imagine the settings are loaded at the application start, so "cancelling the current operation" means exit the application?
Anyway, let's keep DEFAULT_VALUE for the "factory reset".